packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts

```diff
@@ -24,14 +25,22 @@ export const useFluentProvider_unstable = (
 ): FluentProviderState => {
   const parentContext = useFluent();
   const parentTheme = useTheme();
+  const parentOverrides = useOverrides();
 
   /**
    * TODO: add merge functions to "dir" merge,
    * nesting providers with the same "dir" should not add additional attributes to DOM
    * see https://github.com/microsoft/fluentui/blob/0dc74a19f3aa5a058224c20505016fbdb84db172/packages/fluentui/react-northstar/src/utils/mergeProviderContexts.ts#L89-L93
    */
-  const { dir = parentContext.dir, targetDocument = parentContext.targetDocument, theme } = props;
-  const mergedTheme = mergeThemes(parentTheme, theme);
+  const {
+    dir = parentContext.dir,
+    targetDocument = parentContext.targetDocument,
+    theme,
+    overrides_unstable: overrides = {},
+  } = props;
+  const mergedTheme = mergeObjects(parentTheme, theme);
+
+  const mergedOverrides = mergeObjects(parentOverrides, overrides);
```

Comment:
```suggestion
  const mergedTheme = shallowMerge(parentTheme, theme);
  const mergedOverrides = shallowMerge(parentOverrides, overrides);
```

Is it a better name?